com.hazelcast.jet.core.ExecutionLifecycleTest.when_psInitThrows_then_jobFails[useLightJob=true]

_master_ (commit 87b01c11f36fb3afb66c46a2c4ce5fc24c5413a5)

Failed on Zing JDK 8: http://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-ZingJDK8/302/testReport/com.hazelcast.jet.core/ExecutionLifecycleTest/when_psInitThrows_then_jobFails_useLightJob_true_/

Stacktrace:
```
java.lang.ArrayIndexOutOfBoundsException: 1
	at java.util.concurrent.CopyOnWriteArrayList.get(CopyOnWriteArrayList.java:388)
	at java.util.concurrent.CopyOnWriteArrayList.get(CopyOnWriteArrayList.java:397)
	at com.hazelcast.jet.core.ExecutionLifecycleTest.assertPsClosedWithError(ExecutionLifecycleTest.java:755)
	at com.hazelcast.jet.core.ExecutionLifecycleTest.when_psInitThrows_then_jobFails(ExecutionLifecycleTest.java:238)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:115)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:107)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

Standard output:
```
Started Running Test: when_psInitThrows_then_jobFails[useLightJob=true]
07:11:20,836 DEBUG |when_psInitThrows_then_jobFails[useLightJob=true]| - [LightMasterContext] when_psInitThrows_then_jobFails - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Start executing light 0640-b2e5-bf4b-0001, execution graph in DOT format:
digraph DAG {
	"test" [localParallelism=default];
}
HINT: You can use graphviz or http://viz-js.com to visualize the printed graph.
07:11:20,836 DEBUG |when_psInitThrows_then_jobFails[useLightJob=true]| - [LightMasterContext] when_psInitThrows_then_jobFails - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Building execution plan for 0640-b2e5-bf4b-0001
07:11:20,837 DEBUG |when_psInitThrows_then_jobFails[useLightJob=true]| - [LightMasterContext] when_psInitThrows_then_jobFails - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Built execution plans for 0640-b2e5-bf4b-0001
07:11:20,838 DEBUG |when_psInitThrows_then_jobFails[useLightJob=true]| - [InitExecutionOperation] when_psInitThrows_then_jobFails - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Initializing execution plan for job 0640-b2e5-bf4b-0001, execution 0640-b2e5-bf4b-0001 from [127.0.0.1]:5701
07:11:20,839 ERROR |when_psInitThrows_then_jobFails[useLightJob=true]| - [ExecutionContext] when_psInitThrows_then_jobFails - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] job '0640-b2e5-bf4b-0001', execution 0640-b2e5-bf4b-0001 encountered an exception in ProcessorSupplier.close(), ignoring it
java.lang.AssertionError: Close called without init being called on all the nodes. Init count: 1 node count: 2
	at org.junit.Assert.fail(Assert.java:89) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.Assert.assertTrue(Assert.java:42) ~[junit-4.13.2.jar:4.13.2]
	at com.hazelcast.jet.core.TestProcessors$MockPS.close(TestProcessors.java:266) ~[test-classes/:?]
	at com.hazelcast.jet.impl.execution.ExecutionContext.completeExecution(ExecutionContext.java:251) ~[classes/:?]
	at com.hazelcast.jet.impl.JobExecutionService.lambda$completeExecution$559b0d71$1(JobExecutionService.java:458) ~[classes/:?]
	at com.hazelcast.jet.function.RunnableEx.run(RunnableEx.java:31) [classes/:?]
	at com.hazelcast.jet.impl.util.Util.doWithClassLoader(Util.java:469) [classes/:?]
	at com.hazelcast.jet.impl.JobExecutionService.completeExecution(JobExecutionService.java:458) [classes/:?]
	at com.hazelcast.jet.impl.JobExecutionService.runLightJob(JobExecutionService.java:280) [classes/:?]
	at com.hazelcast.jet.impl.operation.InitExecutionOperation.doRun(InitExecutionOperation.java:82) [classes/:?]
	at com.hazelcast.jet.impl.operation.AsyncOperation.run(AsyncOperation.java:53) [classes/:?]
	at com.hazelcast.spi.impl.operationservice.Operation.call(Operation.java:189) [classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.call(OperationRunnerImpl.java:272) [classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:248) [classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:213) [classes/:?]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationExecutorImpl.run(OperationExecutorImpl.java:411) [classes/:?]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationExecutorImpl.runOrExecute(OperationExecutorImpl.java:438) [classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.doInvokeLocal(Invocation.java:600) [classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.doInvoke(Invocation.java:579) [classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.invoke0(Invocation.java:540) [classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.invoke(Invocation.java:240) [classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.InvocationBuilderImpl.invoke(InvocationBuilderImpl.java:59) [classes/:?]
	at com.hazelcast.jet.impl.LightMasterContext.invokeOnParticipant(LightMasterContext.java:199) [classes/:?]
	at com.hazelcast.jet.impl.LightMasterContext.invokeOnParticipants(LightMasterContext.java:185) [classes/:?]
	at com.hazelcast.jet.impl.LightMasterContext.start(LightMasterContext.java:117) [classes/:?]
	at com.hazelcast.jet.impl.JobCoordinationService.submitLightJob(JobCoordinationService.java:292) [classes/:?]
	at com.hazelcast.jet.impl.operation.SubmitJobOperation.doRun(SubmitJobOperation.java:49) [classes/:?]
	at com.hazelcast.jet.impl.operation.AsyncOperation.run(AsyncOperation.java:53) [classes/:?]
	at com.hazelcast.spi.impl.operationservice.Operation.call(Operation.java:189) [classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.call(OperationRunnerImpl.java:272) [classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:248) [classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:213) [classes/:?]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationExecutorImpl.run(OperationExecutorImpl.java:411) [classes/:?]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationExecutorImpl.runOrExecute(OperationExecutorImpl.java:438) [classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.doInvokeLocal(Invocation.java:600) [classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.doInvoke(Invocation.java:579) [classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.invoke0(Invocation.java:540) [classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.invoke(Invocation.java:240) [classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.InvocationBuilderImpl.invoke(InvocationBuilderImpl.java:59) [classes/:?]
	at com.hazelcast.jet.impl.JetInstanceImpl.newLightJobInt(JetInstanceImpl.java:82) [classes/:?]
	at com.hazelcast.jet.impl.AbstractJetInstance.newLightJob(AbstractJetInstance.java:145) [classes/:?]
	at com.hazelcast.jet.core.ExecutionLifecycleTest.newJob(ExecutionLifecycleTest.java:698) [test-classes/:?]
	at com.hazelcast.jet.core.ExecutionLifecycleTest.runJobExpectFailure(ExecutionLifecycleTest.java:712) [test-classes/:?]
	at com.hazelcast.jet.core.ExecutionLifecycleTest.when_psInitThrows_then_jobFails(ExecutionLifecycleTest.java:235) [test-classes/:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0-zing_18.06.0.0]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0-zing_18.06.0.0]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0-zing_18.06.0.0]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0-zing_18.06.0.0]
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59) [junit-4.13.2.jar:4.13.2]
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12) [junit-4.13.2.jar:4.13.2]
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56) [junit-4.13.2.jar:4.13.2]
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17) [junit-4.13.2.jar:4.13.2]
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:115) [test-classes/:?]
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:107) [test-classes/:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0-zing_18.06.0.0]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0-zing_18.06.0.0]
07:11:20,840 DEBUG |when_psInitThrows_then_jobFails[useLightJob=true]| - [JobExecutionService] when_psInitThrows_then_jobFails - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Completed execution of job '0640-b2e5-bf4b-0001', execution 0640-b2e5-bf4b-0001
07:11:20,840 ERROR |when_psInitThrows_then_jobFails[useLightJob=true]| - [InitExecutionOperation] when_psInitThrows_then_jobFails - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] mock error
java.lang.AssertionError: mock error
	at com.hazelcast.jet.core.ExecutionLifecycleTest.<clinit>(ExecutionLifecycleTest.java:108) ~[test-classes/:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0-zing_18.06.0.0]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0-zing_18.06.0.0]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0-zing_18.06.0.0]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0-zing_18.06.0.0]
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.runners.Parameterized$RunnersFactory.allParameters(Parameterized.java:424) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.runners.Parameterized$RunnersFactory.<init>(Parameterized.java:375) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.runners.Parameterized$RunnersFactory.<init>(Parameterized.java:360) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.runners.Parameterized.<init>(Parameterized.java:303) ~[junit-4.13.2.jar:4.13.2]
	at sun.reflect.GeneratedConstructorAccessor31.newInstance(Unknown Source) ~[?:?]
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[?:1.8.0-zing_18.06.0.0]
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423) ~[?:1.8.0-zing_18.06.0.0]
	at org.junit.internal.builders.AnnotatedBuilder.buildRunner(AnnotatedBuilder.java:104) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.internal.builders.AnnotatedBuilder.runnerForClass(AnnotatedBuilder.java:86) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.runners.model.RunnerBuilder.safeRunnerForClass(RunnerBuilder.java:70) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.internal.builders.AllDefaultPossibilitiesBuilder.runnerForClass(AllDefaultPossibilitiesBuilder.java:37) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.runner.Computer.getRunner(Computer.java:50) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.runner.Computer$1.runnerForClass(Computer.java:31) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.runners.model.RunnerBuilder.safeRunnerForClass(RunnerBuilder.java:70) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.runners.model.RunnerBuilder.runners(RunnerBuilder.java:125) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.runners.model.RunnerBuilder.runners(RunnerBuilder.java:111) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.runners.Suite.<init>(Suite.java:81) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.runner.Computer$2.<init>(Computer.java:33) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.runner.Computer.getSuite(Computer.java:28) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.runner.Request.classes(Request.java:77) ~[junit-4.13.2.jar:4.13.2]
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.createRequestAndRun(JUnitCoreWrapper.java:126) ~[surefire-junit47-2.22.2.jar:2.22.2]
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.executeEager(JUnitCoreWrapper.java:107) ~[surefire-junit47-2.22.2.jar:2.22.2]
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:83) ~[surefire-junit47-2.22.2.jar:2.22.2]
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:75) ~[surefire-junit47-2.22.2.jar:2.22.2]
	at org.apache.maven.surefire.junitcore.JUnitCoreProvider.invoke(JUnitCoreProvider.java:158) ~[surefire-junit47-2.22.2.jar:2.22.2]
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384) ~[surefire-booter-2.22.2.jar:2.22.2]
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345) ~[surefire-booter-2.22.2.jar:2.22.2]
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126) ~[surefire-booter-2.22.2.jar:2.22.2]
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418) ~[surefire-booter-2.22.2.jar:2.22.2]
07:11:20,846 DEBUG || - [InitExecutionOperation] hz.crazy_hawking.generic-operation.thread-1 - [127.0.0.1]:5702 [dev] [5.0-SNAPSHOT] Initializing execution plan for job 0640-b2e5-bf4b-0001, execution 0640-b2e5-bf4b-0001 from [127.0.0.1]:5701
07:11:20,847 DEBUG || - [JobExecutionService] hz.crazy_hawking.generic-operation.thread-0 - [127.0.0.1]:5702 [dev] [5.0-SNAPSHOT] job '0640-b2e5-bf4b-0001', execution 0640-b2e5-bf4b-0001 calling completeExecution because execution terminated before it started
07:11:20,847 DEBUG || - [JobExecutionService] hz.crazy_hawking.generic-operation.thread-0 - [127.0.0.1]:5702 [dev] [5.0-SNAPSHOT] Completed execution of job '0640-b2e5-bf4b-0001', execution 0640-b2e5-bf4b-0001
07:11:20,848 DEBUG || - [JobExecutionService] hz.crazy_hawking.generic-operation.thread-1 - [127.0.0.1]:5702 [dev] [5.0-SNAPSHOT] Completed execution of job '0640-b2e5-bf4b-0001', execution 0640-b2e5-bf4b-0001
07:11:20,848 ERROR || - [InitExecutionOperation] hz.crazy_hawking.generic-operation.thread-1 - [127.0.0.1]:5702 [dev] [5.0-SNAPSHOT] mock error
java.lang.AssertionError: mock error
	at com.hazelcast.jet.core.ExecutionLifecycleTest.<clinit>(ExecutionLifecycleTest.java:108) ~[test-classes/:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0-zing_18.06.0.0]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0-zing_18.06.0.0]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0-zing_18.06.0.0]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0-zing_18.06.0.0]
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.runners.Parameterized$RunnersFactory.allParameters(Parameterized.java:424) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.runners.Parameterized$RunnersFactory.<init>(Parameterized.java:375) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.runners.Parameterized$RunnersFactory.<init>(Parameterized.java:360) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.runners.Parameterized.<init>(Parameterized.java:303) ~[junit-4.13.2.jar:4.13.2]
	at sun.reflect.GeneratedConstructorAccessor31.newInstance(Unknown Source) ~[?:?]
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[?:1.8.0-zing_18.06.0.0]
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423) ~[?:1.8.0-zing_18.06.0.0]
	at org.junit.internal.builders.AnnotatedBuilder.buildRunner(AnnotatedBuilder.java:104) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.internal.builders.AnnotatedBuilder.runnerForClass(AnnotatedBuilder.java:86) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.runners.model.RunnerBuilder.safeRunnerForClass(RunnerBuilder.java:70) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.internal.builders.AllDefaultPossibilitiesBuilder.runnerForClass(AllDefaultPossibilitiesBuilder.java:37) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.runner.Computer.getRunner(Computer.java:50) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.runner.Computer$1.runnerForClass(Computer.java:31) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.runners.model.RunnerBuilder.safeRunnerForClass(RunnerBuilder.java:70) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.runners.model.RunnerBuilder.runners(RunnerBuilder.java:125) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.runners.model.RunnerBuilder.runners(RunnerBuilder.java:111) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.runners.Suite.<init>(Suite.java:81) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.runner.Computer$2.<init>(Computer.java:33) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.runner.Computer.getSuite(Computer.java:28) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.runner.Request.classes(Request.java:77) ~[junit-4.13.2.jar:4.13.2]
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.createRequestAndRun(JUnitCoreWrapper.java:126) ~[surefire-junit47-2.22.2.jar:2.22.2]
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.executeEager(JUnitCoreWrapper.java:107) ~[surefire-junit47-2.22.2.jar:2.22.2]
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:83) ~[surefire-junit47-2.22.2.jar:2.22.2]
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:75) ~[surefire-junit47-2.22.2.jar:2.22.2]
	at org.apache.maven.surefire.junitcore.JUnitCoreProvider.invoke(JUnitCoreProvider.java:158) ~[surefire-junit47-2.22.2.jar:2.22.2]
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384) ~[surefire-booter-2.22.2.jar:2.22.2]
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345) ~[surefire-booter-2.22.2.jar:2.22.2]
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126) ~[surefire-booter-2.22.2.jar:2.22.2]
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418) ~[surefire-booter-2.22.2.jar:2.22.2]
07:11:20,854 ERROR || - [LightMasterContext] ForkJoinPool.commonPool-worker-3 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] 0640-b2e5-bf4b-0001 encountered an exception in ProcessorMetaSupplier.complete(), ignoring it
java.lang.AssertionError: all PS that have been init should have been closed at this point expected:<2> but was:<1>
	at org.junit.Assert.fail(Assert.java:89) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.Assert.failNotEquals(Assert.java:835) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.Assert.assertEquals(Assert.java:647) ~[junit-4.13.2.jar:4.13.2]
	at com.hazelcast.jet.core.TestProcessors$MockPMS.close(TestProcessors.java:189) ~[test-classes/:?]
	at com.hazelcast.jet.impl.LightMasterContext.finalizeJob(LightMasterContext.java:129) ~[classes/:?]
	at com.hazelcast.jet.impl.LightMasterContext.lambda$start$1(LightMasterContext.java:118) ~[classes/:?]
	at com.hazelcast.jet.impl.LightMasterContext.lambda$invokeOnParticipant$4(LightMasterContext.java:217) ~[classes/:?]
	at com.hazelcast.spi.impl.AbstractInvocationFuture$WhenCompleteNode.lambda$execute$0(AbstractInvocationFuture.java:1563) ~[classes/:?]
	at com.hazelcast.test.TestLoggingUtils$CustomTestNameAwareForkJoinPool$TestNameAwareRunnable.run(TestLoggingUtils.java:104) [test-classes/:?]
	at java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1402) [?:1.8.0-zing_18.06.0.0]
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289) [?:1.8.0-zing_18.06.0.0]
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056) [?:1.8.0-zing_18.06.0.0]
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692) [?:1.8.0-zing_18.06.0.0]
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157) [?:1.8.0-zing_18.06.0.0]
07:11:20,873  INFO |when_psInitThrows_then_jobFails[useLightJob=true]| - [SimpleTestInClusterSupport] Time-limited test - Ditching 0 jobs in SimpleTestInClusterSupport.@After: []
07:11:20,873  INFO |when_psInitThrows_then_jobFails[useLightJob=true]| - [SimpleTestInClusterSupport] Time-limited test - Destroying 0 distributed objects in SimpleTestInClusterSupport.@After: []
BuildInfo right after when_psInitThrows_then_jobFails[useLightJob=true](com.hazelcast.jet.core.ExecutionLifecycleTest): BuildInfo{version='5.0-SNAPSHOT', build='20210528', buildNumber=20210528, revision=87b01c1, enterprise=false, serializationVersion=1, jet=JetBuildInfo{version='5.0-SNAPSHOT', build='20210528', revision='87b01c1'}}
Hiccups measured while running test 'when_psInitThrows_then_jobFails[useLightJob=true](com.hazelcast.jet.core.ExecutionLifecycleTest):'
07:11:20, accumulated pauses: 847 ms, max pause: 14 ms, pauses over 1000 ms: 0
```

--- a/hazelcast/src/test/java/com/hazelcast/jet/core/ExecutionLifecycleTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/jet/core/ExecutionLifecycleTest.java
@@ -780,7 +780,7 @@ public class ExecutionLifecycleTest extends SimpleTestInClusterSupport {
         assertBetween("close count", MockPS.closeCount.get(), minCount, MEMBER_COUNT);
         assertBetween("received close errors", MockPS.receivedCloseErrors.size(), minCount, MEMBER_COUNT);
 
-        for (int i = 0; i < MEMBER_COUNT; i++) {
+        for (int i = 0; i < MockPS.receivedCloseErrors.size(); i++) {
             assertOneOfExceptionsInCauses(MockPS.receivedCloseErrors.get(i),
                     MOCK_ERROR,
                     new CancellationException(),

